src/problems/RadLineCooling/test_rad_line_cooling.cpp

```diff
@@ -0,0 +1,295 @@
+/// \file test_rad_line_cooling.cpp
+/// \brief Defines a test problem for line cooling and cosmic-ray heating in a uniform medium.
+///
+
+#include <cmath>
+
+#include "AMReX_Array.H"
+#include "AMReX_BC_TYPES.H"
+#include "AMReX_Print.H"
+#include "physics_info.hpp"
+#include "test_rad_line_cooling.hpp"
+#include "util/fextract.hpp"
+
+static constexpr bool export_csv = true;
+
+struct PulseProblem {
+}; // dummy type to allow compile-type polymorphism via template specialization
+
+const double cooling_rate = 0.1;
+const double CR_heating_rate = 0.03;
+
+constexpr double c = 1.0;
+constexpr double chat = c;
+constexpr double v0 = 0.0;
+constexpr double kappa0 = 0.0;
+
+constexpr double T0 = 1.0;   // temperature
+constexpr double rho0 = 1.0; // matter density
+constexpr double a_rad = 1.0;
+constexpr double mu = 1.5; // mean molecular weight; so that C_V = 1.0
+constexpr double C_V = 1.0;
+constexpr double k_B = 1.0;
+
+constexpr double nu_unit = 1.0;
+constexpr double erad_floor = a_rad * 1e-20;
+
+const double max_time = 10.0;
+
+template <> struct SimulationData<PulseProblem> {
```

Comment:
Maybe call this something other than PulseProblem, since it appears to be unrelated to the actual radiation pulse problem described in our papers?